src/views/Log.vue

```diff
@@ -315,20 +371,23 @@ export default {
     }
   },
 
-  created () {
-    // set the ID/file if specified in initialOptions
-    if (this.initialOptions?.tokens?.task) {
-      this.relativeID = this.initialOptions.tokens.relative_id
-      this.jobLog = 1
-    }
-    if (this.initialOptions?.file) {
-      this.file = this.initialOptions.file
-    }
-  },
-
-  async mounted () {
-    // load the log file list and subscribe on startup
-    await this.updateLogFileList()
+  mounted () {
+    // Watch id & file together:
+    this.$watch(
+      () => ({
+        id: this.id ?? undefined, // (treat null as undefined)
+        file: this.file ?? undefined
+      }),
+      async ({ id }, old) => {
+        // update the query when the id or file change
+        this.updateQuery()
+        // refresh the file list when the id changes
+        if (id !== old?.id) {
+          await this.updateLogFileList()
+        }
+      },
+      { immediate: true }
+    )
```

Comment:
Rejigged the watchers for `file` and `id` to prevent race conditions. The usual order of events now is (for the example of opening the workflow log):
1. `id` gets set for the first time to `'~user/workflow'`. `file` is nullish.
    - `updateQuery()` gets called because the `id` changed, but because `file` is nullish it doesn't really do anything https://github.com/MetRonnie/cylc-ui/blob/58f5d96f39f1e9968daa59cdfb6d2fd0c74ef84f/src/views/Log.vue#L429-L431
    - `updateLogFileList()` gets called because the `id` changed
 
1. `file` has been set to the default scheduler log by `updateLogFileList()`
    - `updateQuery()` gets called again because `file` changed, and causes the log file to display
    - `updateLogFileList()` doesn't get called because `id` is unchanged